configure.ac

```diff
@@ -228,6 +229,9 @@ if test "$ac_enable_publish" = "yes"; then
    AC_CHECK_HEADERS(curl.h,[], AC_MSG_ERROR([curl missing]))	
    AC_CHECK_LIB(curl, curl_global_init,, AC_MSG_ERROR([libcurl missing]))
    AC_DEFINE(CLIXON_PUBLISH_STREAMS, 1, [Enable publish of notification streams using SSE and curl])
+   with_curl=yes
```

Comment:
The dependency here is wrong in the original configure.ac.
That is, curl is checked for only if --enable-publish is set.
But --enable-publish is a little-used option while the clxon_util_ applications are necessary for regression testing.
Therefore, the with-curl should be made independent of --enable-publish and also the dependencies on the regression testing should be sorted out, ie the places where clixon_util_stream etc are used in the tests.
My recommendation is to move this out of this PR and add a separate issue on this.

Reply:
Yeah, I wanted to keep the diff as small as possible, but you're right I considered refactoring that whole thing first.  OK, I'll break out a refactor as a separate PR.

Reply:
OK, so this one turned out to be a lot simpler[^1] than I first thought, so I'm including it in the same PR.  The libcurl check only affects `util/clixon_util_stream.c` and `lib/src/clixon_stream.c`, the rest of the test system use the curl tool.

[^1]: Unless I've missed something glaringly obvious.